File: daemon/src/mx/mx_subjects.py
"""
Must match firmware/magic/lib/Mx/mx_subjects.h EXACTLY
"""

SUBJECTS = {
    0x0001: "node_status",
    0x0002: "relay_state",
    0x0003: "sensor_data",
    0x0004: "gps_position",
    0x0010: "command",
    0x0011: "command_reply",
    0x0020: "mesh_neighbor",
    0x0021: "mesh_route",
    0x0030: "ota_announce",
    0x0040: "schedule",
    0x00FF: "heartbeat",
}

BY_NAME = {v: k for k, v in SUBJECTS.items()}

File: firmware/magic/lib/Mx/mx_subjects.cpp
// firmware/magic/lib/Mx/mx_subjects.cpp
#include "mx_message.h"

namespace MxSubjects {
    const char* nameOf(uint16_t id) {
        switch (id) {
            case NODE_STATUS:    return "NODE_STATUS";
            case RELAY_STATE:    return "RELAY_STATE";
            case SENSOR_DATA:    return "SENSOR_DATA";
            case GPS_POSITION:   return "GPS_POSITION";
            case COMMAND:        return "COMMAND";
            case COMMAND_REPLY:  return "COMMAND_REPLY";
            case MESH_NEIGHBOR:  return "MESH_NEIGHBOR";
            case MESH_ROUTE:     return "MESH_ROUTE";
            case OTA_ANNOUNCE:   return "OTA_ANNOUNCE";
            case SCHEDULE:       return "SCHEDULE";
            case HEARTBEAT:      return "HEARTBEAT";
            default:             return "UNKNOWN_SUBJECT";
        }
    }
}
